features/FEATURE_BLE/targets/TARGET_Cypress/TARGET_CYW43XXX/CyVscCmd.cpp

```diff
@@ -0,0 +1,142 @@
+/* mbed Microcontroller Library
+ *
+ * Copyright 2019 Cypress Semiconductor Corporation
+ * SPDX-License-Identifier: Apache-2.0
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#include <mbed.h>
+#include <string.h>
+#include "wsf_types.h"
+#include "wsf_msg.h"
+#include "wsf_buf.h"
+#include "wsf_cs.h"
+#include "hci_api.h"
+#include "CyVscCmd.h"
+
+uint16_t vsc_event_mask = 0;
+wsfQueue_t      vscCmdQueue;
+wsfQueue_t      vscRspQueue;
+static volatile int number_of_msg = 0;
+static int cy_vsc_init = 0;
+
+typedef struct {
+    void        *pNext;
+    uint16_t opcode;
+    cy_vsc_callback_t *vsc_cb;
+} cy_vendor_specific_cmd;
+
+typedef struct {
+    void        *pNext;
+    uint16_t    opcode;
+    uint8_t     status;
+    uint8_t     param_len;
+    uint8_t     *param;
+} cy_vendor_specific_rsp;
+
+Thread threadVsc(osPriorityNormal, OS_STACK_SIZE, NULL, "VSC_CMD");
+
+static cy_vendor_specific_cmd *cy_find_vsc_form_opcode (uint16_t opcode)
+{
+    if(!(WsfQueueEmpty(&vscCmdQueue)))
+    {
+        cy_vendor_specific_cmd *cy_cmd = (cy_vendor_specific_cmd *)WsfQueueDeq(&vscCmdQueue);
+        if( cy_cmd != NULL )
+        {
```

Comment:
if this is Mbed OS implementation, why it does not follow Mbed OS coding style ?

Reply:
Updated